proto/storage/cluster.proto

```diff
@@ -94,6 +95,71 @@ message AdmissionControllerConfig {
   bool enforce_on_updates = 5;
 }
 
+enum CollectorClusterLevelRuntimeFeature {
+  PROCESSES = 0;
+  NETWORK_ENDPOINTS = 2;
+}
+
+enum CollectorNamespaceLevelRuntimeFeature {
+  NETWORK_CONNECTIONS = 0;
+}
+
+message ProcessStatus {
+  bool enabled = 1;
+}
+
+message NetworkConnectionStatus {
+  bool enabled = 1;
+  bool aggregate_external = 2;
+}
+
+message NetworkEndpointStatus {
+  bool enabled = 1;
+  bool processes_listening_on_port = 2;
+}
+
+message CollectorFeatureStatus {
+  oneof status {
+    ProcessStatus process_status = 1;
+    NetworkConnectionStatus network_connection_status = 2;
+    NetworkEndpointStatus network_endpoint_status = 3;
+  }
+}
+
+message CollectorClusterLevelConfig {
+  ProcessStatus process_status = 1;
+  NetworkEndpointStatus network_endpoint_status = 2;
+}
+
+message NamespaceRule {
+  string namespace = 1;
+  storage.MatchType match_type = 2;
+}
+
+message NamespaceSelection {
+  repeated NamespaceRule namespace_rules = 1;
```

Comment:
The set of namespaces in NamespaceSelection will be the union of namespace rules.